src/ServicePulse.Host/vue/src/stores/MonitoringStore.ts

```diff
@@ -85,33 +80,31 @@ export const useMonitoringStore = defineStore("MonitoringStore", () => {
 
   function sortEndpointList() {
     const sortByProperty = sortBy.value;
-    let comparator;
-
-    if (sortByProperty === "name") {
-      comparator = (a, b) => {
-        return isSortAscending.value ? a.name.localeCompare(b.name) : b.name.localeCompare(a.name);
-      };
-    } else {
-      comparator = (a, b) => {
-        const propertyA = getPropertyValue(a, `metrics.${sortByProperty}.average`);
-        const propertyB = getPropertyValue(b, `metrics.${sortByProperty}.average`);
-
-        return isSortAscending.value ? propertyA - propertyB : propertyB - propertyA;
-      };
-    }
+    const comparator = (() => {
+      if (sortByProperty === "name") {
```

Comment:
This can actually be type-safe. The trick is to use [keyof operator](https://www.typescriptlang.org/docs/handbook/2/keyof-types.html).
What it does is that it ensures the `sortByProperty` is a valid prop of `Endpoint`.
To do that, you need to change the type of the ref `const sortBy = ref<keyof Endpoint>("name");`
And the change this functions to be:
```ts
async function updateSort(newSortBy: keyof Endpoint = "name", newIsSortAscending = false) {
    sortBy.value = newSortBy;
    isSortAscending.value = newIsSortAscending;
    await updateEndpointList();
  }
``` 
Now, the consumer of `updateSort` will get an error if they try to pass a prop that does not exist.

All that said, I looked at the `EndpointList.vue` file, and it will be a bit tricky to get this to work properly.
Anyway, at least now you know, you could do it 😄